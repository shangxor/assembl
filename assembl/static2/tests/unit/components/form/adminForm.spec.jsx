import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import AdminForm from '../../../../js/app/components/form/adminForm';

describe('AdminForm component', () => {
  const handleSubmitSpy = jest.fn();

  it('should render an admin form', () => {
    const setRouteLeaveHookSpy = jest.fn();
    const props = {
      handleSubmit: handleSubmitSpy,
      pristine: false,
      routes: [],
      router: {
        setRouteLeaveHook: setRouteLeaveHookSpy
      },
      submitting: false
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<AdminForm {...props}>nothing</AdminForm>);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});