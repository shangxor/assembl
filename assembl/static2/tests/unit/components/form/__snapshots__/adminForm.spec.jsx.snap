// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminForm component should render an admin form 1`] = `
<withRouter(DumbForm)
  handleSubmit={[MockFunction]}
  pristine={false}
  submitting={false}
>
  <SaveButtonInPortal
    saveAction={[MockFunction]}
    specificClasses={null}
  />
  nothing
</withRouter(DumbForm)>
`;
